nsxt/data_source_nsxt_policy_edge_cluster.go

```diff
@@ -20,17 +22,55 @@ func dataSourceNsxtPolicyEdgeCluster() *schema.Resource {
 			"display_name": getDataSourceDisplayNameSchema(),
 			"description":  getDataSourceDescriptionSchema(),
 			"path":         getPathSchema(),
+			"site_path": {
+				Type:         schema.TypeString,
+				Description:  "Path of the site this Transport Zone belongs to",
+				Optional:     true,
+				ValidateFunc: validatePolicyPath(),
+			},
 		},
 	}
 }
 
 func dataSourceNsxtPolicyEdgeClusterRead(d *schema.ResourceData, m interface{}) error {
 	// Read an edge cluster by name or id
-	connector := getPolicyConnector(m)
-	client := enforcement_points.NewDefaultEdgeClustersClient(connector)
+	objSitePath := d.Get("site_path").(string)
 
 	objID := d.Get("id").(string)
 	objName := d.Get("display_name").(string)
+
+	if !isPolicyGlobalManager(m) && objSitePath != "" {
+		return globalManagerOnlyError()
+	}
+	if isPolicyGlobalManager(m) {
+		if objSitePath == "" {
+			return attributeRequiredGlobalManagerError("site_path")
+		}
+
+		query := make(map[string]string)
+		globalPolicyEnforcementPointPath := getGlobalPolicyEnforcementPointPath(m, &objSitePath)
+		query["parent_path"] = globalPolicyEnforcementPointPath
+		obj, err := policyDataSourceResourceRead(d, getPolicyConnector(m), "PolicyEdgeCluster", query)
+		if err != nil {
+			return err
+		}
+		converter := bindings.NewTypeConverter()
+		converter.SetMode(bindings.REST)
+		dataValue, errors := converter.ConvertToGolang(obj, gm_model.PolicyEdgeClusterBindingType())
+		if len(errors) > 0 {
+			return errors[0]
+		}
+		edgeClusterResource := dataValue.(gm_model.PolicyEdgeCluster)
```

Comment:
All these are standard attributes, so these are already set in policyDataSourceResourceRead. Looks like the conversion to specific binding type is not needed here?

Reply:
+1

Reply:
Done